GitUI/Shells/ShellProvider.cs

```diff
@@ -0,0 +1,43 @@
+﻿using ConEmu.WinForms;
+
+namespace GitUI.Shells
+{
+    public class ShellProvider
+    {
+        private static PwshShell PwshShellInstance;
+        private static PowerShellShell PowerShellShellInstance;
+        private static CmdShell CmdShellInstance;
+        private static BashShell BashShellInstance;
+
+        public IShellDescriptor GetShell(string shellType)
+        {
+            switch (shellType.ToLowerInvariant())
+            {
+                case "pwsh":
+                    return PwshShellInstance ?? (PwshShellInstance = new PwshShell());
```

Comment:
The shell classes should provide the shell name as a static member.
```suggestion
                case PwshShell.Name:
                    return PwshShellInstance ?? (PwshShellInstance = new PwshShell());
```